crates/common/src/term.rs

```diff
@@ -159,6 +159,16 @@ impl Reporter for SpinnerReporter {
             version.minor,
             version.patch
         ));
+
+        if foundry_common::shell::verbosity() > 0 {
```

Comment:
Awesome! I'll go with this. Thanks!

Reply:
@grandizzy after implementing it this way, there's still some interference between the spinner and the `dirty_files` list.
```
/home/mg/Code/foundry-test-project/lib/forge-std/src/safeconsole.sol
/home/mg/Code/foundry-test-project/lib/forge-std/src/StdUtils.sol
/home/mg/Code/foundry-test-project/script/Counter.s.sol
/home/mg/Code/foundry-test-project/lib/forge-std/src/StdError.sol
[⠑] Compiling filesCode/foundry-test-project/lib/forge-std/src/StdCheats.sol
/home/mg/Code/foundry-test-project/lib/forge-std/src/console.sol
/home/mg/Code/foundry-test-project/lib/forge-std/src/StdChains.sol
/home/mg/Code/foundry-test-project/lib/forge-std/src/StdMath.sol
/home/mg/Code/foundry-test-project/lib/forge-std/src/Vm.sol
```

The spinner displayed by the `SpinnerReporter` shows up in the middle of the file list.
This seems to be because `SpinnerReporter::spawn()` is called before the compilation process starts, in `crates/common/src/compile.rs:477`:

```
/// Configures the reporter and runs the given closure.
pub fn with_compilation_reporter<O>(quiet: bool, f: impl FnOnce() -> O) -> O {
    #[allow(clippy::collapsible_else_if)]
    let reporter = if quiet {
        Report::new(NoReporter::default())
    } else {
        if std::io::stdout().is_terminal() {
            Report::new(SpinnerReporter::spawn())
        } else {
            Report::new(BasicStdoutReporter::default())
        }
    };

    foundry_compilers::report::with_scoped(&reporter, f)
}
```

The hook in which the `dirty_files` are available to be printed out is `on_compiler_spawn` in `crates/common/src/term.rs:158` and it only runs after the spinner is spawned.

Reply:
There shouldn't be an if / else, just if verbose print dirty files. The other should always be printed when compiler spawned.

Reply:
`on_compiler_spawn` looks like the best place to output the dirty files list, but since it necessarily runs after the spinner reporter is spawned I'm wondering if it makes sense doing it somewhere else.

Reply:
> yeah, what I meant is that this else should be removed and always send 2nd message, regardless verbosity
> 
> https://github.com/foundry-rs/foundry/blob/c8a3362169891c575a674ff31b43e909b9e17b72/crates/common/src/term.rs#L168

Got it, thanks. This one's been addressed.

Reply:
:thinking: I think the message interfering with it is the `[⠑] Compiling files` displayed for the spinner every time it ticks. I still see it happen after removing the `if..else`:
```
/home/mg/Code/mgiagante/foundry-test-project/lib/forge-std/src/interfaces/IERC20.sol
[⠑] Compiling filesante/foundry-test-project/lib/forge-std/src/interfaces/IMulticall3.sol
/home/mg/Code/mgiagante/foundry-test-project/script/Counter.s.sol
```

Reply:
the verbose message
https://github.com/foundry-rs/foundry/blob/6b64f2b3864d6e7addbb566a4b7df88085823ca6/crates/common/src/term.rs#L168-L178
should be sent first, like

```rust
        if foundry_common::shell::verbosity() > 0 {
            self.send_msg(format!(
                "Compiling files\n{}",
                dirty_files
                    .iter()
                    .map(|path| path.display().to_string())
                    .collect::<Vec<String>>()
                    .join("\n")
            ));
        }
        self.send_msg(format!(
            "Compiling {} files with {} {}.{}.{}",
            dirty_files.len(),
            compiler_name,
            version.major,
            version.minor,
            version.patch
        ));
```